### PR TITLE
Add Observability to Document Model Operations is DynamoDB HLL

### PR DESCRIPTION
**Pull request by 96malhar**
## Description

This PR adds observability to DocumentModel operations is DynamoDB HLL by creating traces using the SDK observability APIs.
* Traces are only created for public APIs that make a network call.
* I have updated the Async public APIs to await the `Task` instead of returning it. This is done to ensure that the trace is not disposed before the function completes its execution.

## Motivation and Context

This change is required as part of the Observability initiative for the SDK and HLLs
DOTNET-7679

## Testing

* All existing Unit and Integrations tests pass locally
* **Pending a full dry-run**. Since I am targeting a feature branch, I'll perform a full dry run on the feature branch itself after observability updates are merged in for both the data model and document model operations

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: sdk/src/Services/DynamoDBv2/Custom/DocumentModel/_async/DocumentBatchGet.Async.cs [hunk 1]
```diff
@@ -22,6 +22,8 @@
 using Amazon.DynamoDBv2.Model;
 using Amazon.Runtime;
 using Amazon.Runtime.Internal;
+using Amazon.Runtime.Telemetry.Tracing;
+using AWSSDK.DynamoDBv2;
```
Comment (normj) [inline sdk/src/Services/DynamoDBv2/Custom/DocumentModel/_async/DocumentBatchGet.Async.cs:26]: I don't understand where this namespace is coming from. There shouldn't be a namespace called "AWSSDK.DynamoDBv2"

File: sdk/src/Services/DynamoDBv2/Custom/DocumentModel/_async/DocumentBatchWrite.Async.cs [hunk 2]
```diff
@@ -66,7 +72,11 @@ public partial class MultiTableDocumentBatchWrite
         /// <returns>A Task that can be used to poll or wait for results, or both.</returns>
         public Task ExecuteAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
-            return ExecuteHelperAsync(cancellationToken);
+            var operationName = DynamoDBTelemetry.ExtractOperationName(nameof(MultiTableDocumentBatchWrite), nameof(ExecuteAsync));
+            using (DynamoDBTelemetry.CreateSpan(TracerProvider, operationName, spanKind: SpanKind.CLIENT))
+            {
+                return ExecuteHelperAsync(cancellationToken);
```
Comment (normj) [inline sdk/src/Services/DynamoDBv2/Custom/DocumentModel/_async/DocumentBatchWrite.Async.cs:78]: Do we need to switch this to use `async` / `await` to maintain trace context? I thought we had to do this for the other PRs. If so there are a few places like this that need to be fixed up.

**Review by normj — CHANGES_REQUESTED** [review]

File: sdk/src/Services/DynamoDBv2/Custom/DocumentModel/_async/DocumentBatchGet.Async.cs (hunk 1 above)
Comment (96malhar) [reply sdk/src/Services/DynamoDBv2/Custom/DocumentModel/_async/DocumentBatchGet.Async.cs:26]: I had added the new internal static class `DynamoDBTelemetry` in the `AWSSDK.DynamoDBv2` namespace. I have changed the namespace for that class to `Amazon.DynamoDBv2` which is an existing one.

File: sdk/src/Services/DynamoDBv2/Custom/DocumentModel/_async/DocumentBatchWrite.Async.cs (hunk 2 above)
Comment (96malhar) [reply sdk/src/Services/DynamoDBv2/Custom/DocumentModel/_async/DocumentBatchWrite.Async.cs:78]: I made this change for most of the async APIs but missed a few spots. Fixed them.

File: sdk/src/Services/DynamoDBv2/Custom/DocumentModel/Table.cs [hunk 3]
```diff
@@ -841,18 +898,23 @@ public static bool TryLoadTable(IAmazonDynamoDB ddbClient,
...
-            }
-            catch
+            var operationName = DynamoDBTelemetry.ExtractOperationName(nameof(Table), nameof(TryLoadTable));
+            var tracerProvider = ddbClient.Config.TelemetryProvider.TracerProvider;
+            using (DynamoDBTelemetry.CreateSpan(tracerProvider, operationName, spanKind: SpanKind.CLIENT))
             {
-                table = null;
-                return false;
+                if (config == null)
+                    throw new ArgumentNullException("config");
```
Comment (normj) [inline sdk/src/Services/DynamoDBv2/Custom/DocumentModel/Table.cs:906]: nit: use `nameof(config)` instead of using string "config"

**Review by normj — APPROVED** [review]

File: sdk/src/Services/DynamoDBv2/Custom/DocumentModel/Table.cs (hunk 3 above)
Comment (96malhar) [reply sdk/src/Services/DynamoDBv2/Custom/DocumentModel/Table.cs]: This is existing code that was just placed inside the `using` scope. I fixed it

**Review by philasmar — APPROVED** [review]

**Pull request merged**